scripts/release/asset_publish.py

```diff
@@ -119,13 +120,53 @@ def model_prepare(model_config: assets.ModelConfig, spec_file_path: Path, model_
                     logger.log_error(f"Error loading flavors from MLmodel file at: {mlmodel_file_path} => {str(e)}")
             can_publish_model = dump_model_spec(model, spec_file_path)
 
+
     else:
         print(model_config.type.value, assets.ModelType.MLFLOW)
         can_publish_model = False
         logger.log_error(f"Model type {model_config.type} not supported yet")
 
     return can_publish_model
 
+def dump_data_spec(data, spec_file) -> bool:
+    """Update the yaml file after getting the data has been prepared."""
+    try:
+        data_dict = json.loads(json.dumps(data._to_dict()))
```

Comment:
Why dump an object to JSON just to parse it again?

Reply:
Unfortunately Yes .
After dumping it to json ,currently we are calling a function create_command, which need file_path, where all specification will be saved. So thats why dumping is important.
Other option is to alter/update the create_command function, so that it can create a command without file path as well, but will it be beneficial?Thoughts?

Reply:
I was referring to the call to `json.dumps()` and the subsequent call to `json.loads()`, which you shouldn't have to do. You're already getting some type of dictionary object from `data._to_dict()`, which should be very similar to the final output of `json.loads(json.dumps(data._to_dict()))` except that perhaps some datatypes are converted in the process. I think it would be better to make those conversions explicit in this class instead.

Reply:
Thankss ! I misunderstood it . 
Done